Extending packer updates (#10599)
hashicorp/packer: website/content/guides/1.7-plugin-upgrade/index.mdx

```

The implementation is extremely similar, but now we use the "NewSet" function to create the plugin Server, and call Run() instead of Serve().

> With the multi-plugin binary example, it is important to not that the name of the binary must be in the form of `packer-pluign-bar`. Otherwise Packer will not register it as a multiple plugin.

Reviewer: ```suggestion
With the multi-plugin binary example, it is important to note that the name of the binary must be in the form of `packer-pluign-name`. Otherwise Packer will not register it as a multi-plugin.
```

```go
go build -o packer-plugin-bar
```

> Then you'd install it by putting it into the plugin directory with the name `packer-plugin-bar`.

Reviewer 2: ```suggestion
Then you'd install it by putting it into the plugin directory with the name `packer-plugin-name`.
```

Reviewer 2: ( ah ditto throughout; I wont add those to avoid noise )

With the multi-plugin binary you would then refer to your newly installed plugin as `bar-foo` within your Packer template. Whereas, in the single binary setup, you'd have used the name "bar" in your Packer template.

> At this point it is important to note that multi-plugins are to Packer, what providers are to Terraform. Multi-plugin binaries are meant to provide one or more plugin types to Packer. See [registering multiple plugins](#registering-multiple-plugins) for details on how to register more than one plugin type.

Reviewer: ```suggestion
At this point it is important to note that multi-plugins are to Packer, what providers are to Terraform. Multi-plugin binaries are meant to provide one or more plugin types to Packer which depend on the same cloud or other APIs. See [registering multiple plugins](#registering-multiple-plugins) for details on how to register more than one plugin type.
```

Reviewer 2: If we add this we should we also probably add a note that it's not possible to configure plugins.( read this [proposal](https://github.com/hashicorp/terraform-proposals/blob/main/module-provider-config/module-providers.md) for more info on what I meant. )

Author: Good call out @azr let me see how to word this. I also wonder if this is text that we should put on the installing plugins docs as the ability to configure plugins in more in tune with the required_plugins block. At least that's my understanding.

> If your plugin providers a single plugin type please continue reading for instruction on using the Set architecture for single plugins.

Reviewer 3: Nice! I was missing this piece while I was reading this and there it is :D

You can still use the `DEFAULT_NAME` constant with a set implementation; in this example, whatever you register with the DEFAULT_NAME will be referenced in the Packer template as "vsphere". You can only register one plugin per plugin set using the DEFAULT_NAME

## Distributing migrated plugins
Once a plugin has been migrated to use the `packer-plugin-sdk` it can be released as it normally would and used by Packer by [installing the plugin](/docs/plugins#installing-plugins) manually into the respective Packer plugin directory.

> If however, you would like to take advantage of Packer's automated plugin installation process via `packer init` -- the preferred method of installation -- you will need to make the plugin available on GitHub and in a repository named after the multi-plugin `https://github.com/<yourorg>/packer-plugin-<name>`. See [Creating a GitHub Release](/docs/plugins/creation#creating-a-github-release) for details on the recommended practice for releasing Packer plugins on GitHub.

Reviewer 3: Awesome!
